realm/src/main/java/io/realm/Realm.java

```diff
@@ -194,10 +194,14 @@ public static Realm getInstance(Context context) {
      * model classes or version has has changed so a migration is required.
      */
     public static Realm getDefaultInstance() {
-        if (defaultConfiguration == null) {
+        final RealmConfiguration configuration;
+        synchronized (Realm.class) {
+            configuration = defaultConfiguration;
+        }
```

Comment:
If the `create(configuration)` is not in the `synchronized` block, i don't really think this fix has too much difference compared with before. Since before calling create, there is still a chance for `setDefaultConfiguration` to change the default configuration. And a reference to the old config will be passed to the `create`. And this would be the exact same behaviour as before.

We might need to think about what do we want to achieve here? Do we want to allow user to call setDefaultConfiguration twice?